src/js/components/JSONEditor.js

```diff
@@ -226,7 +226,16 @@ class JSONEditor extends React.Component {
     const lastError = this.jsonError;
 
     // Calculate what the next state is going to be
-    const { jsonValue, jsonMeta, jsonError } = this.getNewJsonState(jsonText);
```

Comment:
would providing a default value in `handleChange` do as well?

`  handleChange(jsonText = "") {` or 

```
  handleChange(jsonText) {
    if (typeof jsonText != "string") {
      jsonText = ""
    }
    ...
```

Reply:
This might work. Do you think the current solution is lacking in some way?

Reply:
yes, i do think it is lacking in clarity. 

the reason is that `getNewJsonState` is quite convoluted. it's either returning the object you resembled or a `null`.

i'd now expect consumers to distinct between the object and `null`.

so we could write 
```
const newJsonState = this.getNewJsonState(jsonText);
if (newJsonState === null) {
  return;
}
const { jsonValue, jsonMeta, jsonError } = newJsonState;
...
```

so we would need to introduce 4 lines to properly and explicitly handle a `null` (which is mentioned in the docs of `getNewJsonState`).

instead we now provide an empty default object to destruct it and return in case each value has not changed between lines 230 and 236. while this is relatively simple to see here it still requires a lot more mental overhead than necessary to understand the intention of the author here.

------------

IF we ever want to fix the root of that complication we'd need to narrow down the interface of `getNewJsonState` from `getNewJsonState(jsonText: string | null): ThatObject | null` to `getNewJsonState(jsonText: string): ThatObject`. 

To do that we should handle `null`s outside of that function and that's why i asked the original question.

Reply:
Okay, I agree with you. I will change it.